src/tree.rs

```diff
@@ -0,0 +1,59 @@
+//! The baseline requirements of any UI Tree so Taffy can efficiently calculate the layout
+
+use crate::{
+    layout::{Cache, Layout},
+    prelude::*,
+};
+
+/// Any item that implements the LayoutTree can be layed out using Taffy's algorithms.
+///
+/// Generally, Taffy expects your Node tree to be indexable by stable indicies. A "stable" index means that the Node's ID
+/// remains the same between re-layouts.
+pub trait LayoutTree {
+    /// Get the list of children IDs for the given node
+    fn children(&self, node: Node) -> &[Node];
+
+    /// Get a specific child of a node, where the index represents the nth child
+    fn child(&self, node: Node, index: usize) -> Node;
+
+    /// Get any available parent for this node
+    fn parent(&self, node: Node) -> Option<Node>;
+
+    // todo: allow abstractions over this so we don't prescribe how layout works
+    // for reference, CSS cascades require context, and storing a full flexbox layout for each node could be inefficient
+    //
+    /// Get the [`FlexboxLayout`] for this Node.
+    fn style(&self, node: Node) -> &FlexboxLayout;
```

Comment:
I would like to suggest that `FlexboxLayout` is renamed to `FlexboxStyle`: the idea being to establish a naming convention that `*Style` is input and `*Layout` is output. `FlexboxLayout` is already in a file called `style.rs` and returned from a method called `style`, so this seems to improve naming consistency.

Reply:
This makes a lot of sense, it is what it should have been all along.

Reply:
I'm on board with that. IMO we should do it in a small seperate PR though.

Reply:
Yes it is, but my point is that the user will have to implement it anyway, so it probably should be reused; so having a `style()` method to ask the user how it resolved a particular `Node`'s properties looks good to me.